### PR TITLE
Fix broken emoji (Safari, Firefox)

### PR DESCRIPTION
**Pull request by thesahindia**
### Details

This PR fixes two issues -
- Broken emoji ui at default skin tone selector (safari)
Screenshot 2022-06-28 at 7 59 10 PM

- Black outlined emoji in Firefox (ubuntu/windows)
Screenshot 2022-06-28 at 7 58 17 PM

### Fixed Issues

$ https://github.com/Expensify/App/issues/9123

### Tests

1. Open Expensify at Safari 
2. Open emoji picker menu
3. Select a skin tone (other than default yellow)
4. Verify the hand emoji isn't broken

---

1. Open Expensify at FireFox (windows/Ubuntu)
2. Open emoji picker menu
3. Verify the hand emoji is colored (not black and white)

---

1. Open Expensify at FireFox (windows/Ubuntu)
2. Add hand emoji (🖐) to the message composer
3. Add hand emoji to the edit comment composer
4. Verify the hand emoji is colored (not black and white)

---
1. Open Expensify at FireFox (windows/Ubuntu)
2. Send hand emoji (🖐)
3. Send hand emoji with some text
4. Send hand emoji with some link
5. Verify the hand emoji is colored (not black and white) in all the messages

- [x] Verify that no errors appear in the JS console

#### Contributor (PR Author) Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] Any functional components have the `displayName` property
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose and it is
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn’t already exist
    - [x] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

<details>
<summary><h4>PR Reviewer Checklist</h4></summary>

- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

</details>

### QA Steps

1. Open Expensify at Safari 
5. Open emoji picker menu
6. Select a skin tone (other than default yellow)
8. Verify the hand emoji isn't broken

---

1. Open Expensify at FireFox (windows/Ubuntu)
2. Open emoji picker menu
3. Verify the hand emoji is colored (not black and white)

---
1. Open Expensify at FireFox (windows/Ubuntu)
2. Add hand emoji (🖐) to the message composer
3. Add hand emoji to the edit comment composer
4. Verify the hand emoji is colored (not black and white)

---
1. Open Expensify at FireFox (windows/Ubuntu)
2. Send hand emoji (🖐)
3. Send hand emoji with some text
4. Send hand emoji with some link
5. Verify the hand emoji is colored (not black and white) in all the messages

- [x] Verify that no errors appear in the JS console

#### Web

Screenshot 2022-06-28 at 8 12 25 PM

#### Mobile Web

Screenshot 2022-06-28 at 8 11 40 PM

#### Desktop

Screenshot 2022-06-28 at 8 18 45 PM

#### iOS

Screenshot 2022-06-28 at 8 11 06 PM

#### Android

Screenshot 2022-06-28 at 8 10 36 PM

### CODE REVIEW

File: src/styles/emojiFont/index.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+const emojiFont = 'Segoe UI Emoji, Noto Color Emoji';
```
Comment (parasharrajat) [inline src/styles/emojiFont/index.js:1]: Lets move the emojiFont under `styles/fontFamily` folder and rename it to `emoji` only.

File: src/styles/emojiFont/index.ios.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+// Using system fonts e.g. Segoe UI Emoji results in error in ios
```
Comment (parasharrajat) [inline src/styles/emojiFont/index.ios.js:1]: Do we need to change the font for Android? If not, rename this file to `index.native.js`. Remove this comment.

File: src/styles/emojiFont/index.ios.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+// so using GTAmericaExp-Regular
+const emojiFont = 'GTAmericaExp-Regular';
```
Comment (parasharrajat) [inline src/styles/emojiFont/index.ios.js:3]:
```suggestion
const EMOJI_TEXT_FONT = 'GTAmericaExp-Regular';
```

File: src/styles/emojiFont/index.js (hunk 1 above)
Comment (parasharrajat) [inline src/styles/emojiFont/index.js:1]: Add a comment on the top saying why do we have to use these fonts for web| Desktop?

**Review by parasharrajat — CHANGES_REQUESTED** [review]

**Comment by thesahindia** [comment]
Made the requested changes.
cc: @parasharrajat

**Review by parasharrajat — CHANGES_REQUESTED** [review]
The big issue: Emojis are different on messages than the picker. 
image

try these emojies.

Picker 
image

**Comment by thesahindia** [comment]
Will look into it.

**Comment by thesahindia** [comment]
We can see this issue on composer too and to fix that we will have to use the fallback fonts. 
For the messages we can add the font here - https://github.com/Expensify/App/blob/eb54f232638d6a41db9170de591506d464941850/src/pages/home/report/ReportActionItemFragment.js#L109
And for composer

**Comment by parasharrajat** [comment]
How are you passing the fallback fonts?

**Comment by thesahindia** [comment]
@parasharrajat, I tried using  `${fontFamily.GTA}, ${fontFamily.EMOJI_TEXT_FONT}` on line 136 
https://github.com/Expensify/App/blob/eb54f232638d6a41db9170de591506d464941850/src/styles/styles.js#L133-L136

but it didn't work. I also tried applying only `fontFamily.EMOJI_TEXT_FONT` ( contains two fonts ) and it also failed. 
It only works when only one font is used.

**Comment by parasharrajat** [comment]
Which platform has issues? Could you please share some videos of where it worked including the fix and before the fix?

**Comment by thesahindia** [comment]
Asked for the help on slack. Here's the [thread](https://expensify.slack.com/archives/C01GTK53T8Q/p1658231631715389).

File: src/components/HTMLEngineProvider/BaseHTMLEngineProvider.js [hunk 4]
```diff
@@ -59,6 +59,7 @@ const BaseHTMLEngineProvider = (props) => {
             tagsStyles={styles.webViewStyles.tagStyles}
             enableCSSInlineProcessing={false}
             systemFonts={_.values(fontFamily)}
+            fallbackFonts={{'GTAmericaExp-Regular': fontFamily.EMOJI_TEXT_FONT}}
```
Comment (parasharrajat) [inline src/components/HTMLEngineProvider/BaseHTMLEngineProvider.js:62]: Hmm..

**Review by parasharrajat — COMMENTED** [review]
Will test it soon.

**Review by parasharrajat — CHANGES_REQUESTED** [review]
Good work, it is working great. 
The only difference is that emoji rendered on each platform has different styles which are expected as we are using native Unicode chars that depend on the platform.

File: src/components/HTMLEngineProvider/BaseHTMLEngineProvider.js (hunk 4 above)
Comment (Beamanator) [reply src/components/HTMLEngineProvider/BaseHTMLEngineProvider.js:62]: I like the idea of putting this in a const in this file, and putting the comment right about that

**Review by Beamanator — CHANGES_REQUESTED** [review]
A few little comments, overall looking fantastic

**Comment by thesahindia** [comment]
> A few little comments, overall looking fantastic

Will make the changes in 2-3 days.
Visiting family :)

**Comment by Beamanator** [comment]
@thesahindia thanks for making the requested changes 7 days ago, would you please tag us when this is ready for review again so we don't delay so long? :D

Also since it has been 7 days since your last commit, would you please merge main & test again to make sure there's no regressions?

**Comment by thesahindia** [comment]
Done 🚀 
cc: @parasharrajat @Beamanator

**Review by parasharrajat — DISMISSED** [review]
Just a small suggestion.

cc: @Beamanator 

#### PR Reviewer Checklist
- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/docs/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/docs/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/docs/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/docs/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] Any functional components have the `displayName` property
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

:ribbon: :eyes: :ribbon:  C+ reviewed

**Comment by thesahindia** [comment]
@Beamanator It's ready for the review now.

**Review by Beamanator — APPROVED** [review]
Let's get this merged :D

**Comment by Beamanator** [comment]
Uhh not emergency, all tests were passing 🤔 I don't believe there were any failing tests when this was merged

**Pull request merged**